chore: UTC-324: [FE] Refactor the CurrentUser provider

nass600 opened the pull request:
Refactored currentUser hooks and atoms into the new AuthProvider for consolidation

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nass600 committed:
chore: UTC-324: [FE] Refactor the CurrentUser provider

nass600 committed:
refactored user atom

nass600 committed:
restoring isLoading

nass600 committed:
last details

nass600 committed:
Merge remote-tracking branch 'origin/develop' into fb-utc-324

yyassi-heartex commented on web/apps/labelstudio/src/components/Menubar/Menubar.jsx:
```diff
@@ -57,7 +57,7 @@ const RightContextMenu = ({ className, ...props }) => {
   const menuDropdownRef = useRef();
   const useMenuRef = useRef();
-  const { user, fetch, isInProgress } = useCurrentUser();
+  const { user, isLoading } = useAuth();
```
I see fetch is being removed with no replacement? was it not being used?

nass600 replied:
exactly

nass600 commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18399644564)

nass600 committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18399644564

nass600 committed:
Merge branch 'develop' into 'fb-utc-324'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18399644564
